tgui/packages/tgui/interfaces/TelecommsMonitor.tsx

```diff
@@ -0,0 +1,162 @@
+import { useBackend, useLocalState } from '../backend';
+import { Section, Stack, Input, Button, Table, LabeledList, NoticeBox } from '../components';
+import { Window } from '../layouts';
+
+enum Screen {
+  Main,
+  Machine,
+}
+
+type Data = {
+  screen: Screen;
+  network: string;
+  error_message: string;
+  machinery: Machine[];
+  machine: Machine;
+};
+
+type Machine = {
+  ref: string;
+  id: string;
+  name: string;
+  linked_machinery: LinkedMachinery[];
+};
+
+type LinkedMachinery = {
+  ref: string;
+  id: string;
+  name: string;
+};
+
+const MachineScreen = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { network, machine } = data;
+  return (
+    <Stack fill vertical>
+      <Stack.Item>
+        <Section
+          title="Entity Information"
+          buttons={
+            <Button
+              content="Main Menu"
+              icon="home"
+              onClick={() => act('return_home')}
+            />
+          }>
+          <LabeledList>
+            <LabeledList.Item label="Network">{network}</LabeledList.Item>
+            <LabeledList.Item label="Network Entity">
+              {machine.name}
+            </LabeledList.Item>
+          </LabeledList>
+        </Section>
+      </Stack.Item>
+      <Stack.Item grow>
+        <Section fill scrollable title="Linked Entities">
+          <Table>
+            <Table.Row header>
+              <Table.Cell>Address</Table.Cell>
+              <Table.Cell>Name</Table.Cell>
+              <Table.Cell>Machine</Table.Cell>
+            </Table.Row>
+            {machine.linked_machinery?.map((lm) => (
```

Comment:
please, prefer these destructured outside of jsx
` const { linked_machinery = [] } = machine`